app/views/pages/show.html.erb

```diff
@@ -18,7 +18,19 @@
     <div id="attachment">
       <% if @page.file.attached? %>
         <%= attachment_icon(@page.file_blob.content_type) %>
-        <p><%= @page.file_blob.filename %></p>
+        <p class="filename"><%= @page.file_blob.filename %></p>
+
+        <%= content_tag :p, class: 'filename-edit-container hidden' do -%>
+          <%= text_field_tag 'filename', @page.file_blob.filename.base %>
+
+          <span id="file-extension">
+            <%= ".#{@page.file_blob.filename.extension}" %>
+          </span>
+
+          <button class="button" type="button" name="update-filename-button" onclick="updateFileName()">
```

Comment:
I was thinking that pressing "Enter" and/or clicking out of the text area could be better. What do you think?

Reply:
I thought of a more visual way on doing so, so the button came to my mind, but I can add an event listener to the text input too. On the other hand, I don't think updating on blur is very friendly on the UX

Reply:
Add just the "Enter" option. I'll do some UX tests with friends and I'll change it if needed! Thanks.